driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -692,6 +695,9 @@ else if (fetchSize != Integer.MAX_VALUE)
    * and handle host failover.
    */
   void execute(final RequestHandler.Callback callback, final Statement statement) {
+    if (cluster.isClosed()) {
```

Comment:
It's also possible that the Session itself is closed, so we should check that too and surface a specific error in that case.

The error message could be something like "Could not send request, <X> is closed." where X is either 'cluster' or 'session'.